scripts/host-check

```diff
@@ -1567,30 +1522,64 @@ def run_checks_specific_platform(
     :return:
         Whether all checks that were performed succeeded.
     """
+    extras_supported: Set[str] = set()
     extras_not_supported: Set[str] = set()
+    extras_errored: Set[str] = set()
+
     print_heading(f"Platform checks - {platform}")
-    platform_success = perform_checks(
-        BASE_CHECKS + PLATFORM_CHECKS_MAP[platform]
-    )
+    plat_results = perform_checks(BASE_CHECKS + PLATFORM_CHECKS_MAP[platform])
+    plat_failure = any(check.is_failure() for check in plat_results.values())
+    plat_error = any(c is CheckState.ERROR for c in plat_results.values())
 
-    if extra_checks is not None:  # check if any extra checks were specified
-        extras_supported, extras_not_supported = run_extra_checks(extra_checks)
+    if extra_checks:
+        print("")  # insert a newline
+        print_heading("Extra checks")
+        for extra_check in sorted(extra_checks):
+            print_subheading(f"{extra_check} checks")
+            extra_results = perform_checks(EXTRA_CHECKS_MAP[extra_check])
+
+            if any(check.is_failure() for check in extra_results.values()):
+                extras_not_supported.add(extra_check)
+            elif any(c is CheckState.ERROR for c in extra_results.values()):
+                extras_errored.add(extra_check)
+            else:
+                extras_supported.add(extra_check)
 
-    print(f"\n{DOUBLE_DASHED_LINE}")
-    if platform_success:
+    # Print a summary of what is and is not supported
+    print_without_newline(f"\n{DOUBLE_DASHED_LINE}")
```

Comment:
Why without newline and then all of the following need to have a `\n` at the beginning of what they print? Same in multiple places below.